src/commands/PublishCommand.js

```diff
@@ -292,7 +292,7 @@ export default class PublishCommand extends Command {
       const version = this.updatesVersions[depName];
 
       if (deps[depName] && version) {
-        deps[depName] = "^" + version;
+        deps[depName] = this.flags.exact ? version : "^" + version;
```

Comment:
I think you might have read it wrong. It's just a ternary operator

```
const a = someFlag ? 1 : 2;
```

Sets a to 1 if someFlag is true and 2 if it's false.